packages/expo-updates/ios/EXUpdates/Logging/UpdatesLogEntry.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+import Foundation
+
+/**
+ Schema for the fields in expo-updates log message JSON strings
+ */
+public struct UpdatesLogEntry: Codable {
+  var timestamp: UInt // seconds since 1/1/1970 UTC
+  var message: String
+  var code: String // One of the UpdatesErrorCode string values
+  var level: String // One of the ExpoModulesCore.LogType string values
+  var updateId: String? // EAS update ID, if any
+  var assetId: String? // EAS asset ID, if any
+  var stacktrace: [String]? // Stacktrace (for error and fatal logs)
+
+  /**
+   Returns a JSON string representation from this UpdatesLogEntry object
+   */
+  public func asString() -> String? {
+    do {
+      let jsonEncoder = JSONEncoder()
+      let jsonData = try jsonEncoder.encode(self)
+      return String(data: jsonData, encoding: .utf8)
+    } catch {
+      return nil
+    }
+  }
+
+  /**
+   Returns a Dictionary representation from this UpdatesLogEntry object
+   */
+  public func asDict() -> [String: Any] {
+    var result = [String: Any]()
+    result["timestamp"] = timestamp
+    result["message"] = message
+    result["code"] = code
+    result["level"] = level
+    if (updateId != nil) {
+      result["updateId"] = updateId
+    }
+    if (assetId != nil) {
+      result["assetId"] = assetId
+    }
+    if stacktrace != nil {
+      let nsstack = NSMutableArray()
+      for s in stacktrace! {
+        nsstack.add(s)
+      }
+      result["stacktrace"] = nsstack
+    }
+    return result
+  }
+
+  /**
+   Returns a new UpdatesLogEntry from a JSON string, or nil if a decoding error occurs
+   */
+  public static func create(from: String) -> UpdatesLogEntry? {
+    do {
+      let jsonDecoder = JSONDecoder()
+      guard let jsonData = from.data(using: .utf8) else { return nil }
+      return try jsonDecoder.decode(UpdatesLogEntry.self, from: jsonData)
+    } catch {
+      return nil
+    }
+  }
+
+  /**
+   Utility method to construct stacktrace as a string array for log entries
+   */
+  private static let STACKTRACE_MAX_LENGTH = 20
```

Comment:
I guess you meant to put this comment a few lines below to refer to `currentStackTrace` instead of the `STACKTRACE_MAX_LENGTH` constant 😛

Reply:
Fixed